src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1428,6 +1428,40 @@ private void ImportCpOpj(int token)
 
         private void ImportUnbox(int token, ILOpcode opCode)
         {
+            TypeDesc type = ResolveTypeToken(token);
+            if (type.IsNullable)
+                throw new NotImplementedException();
+
+            LLVMValueRef eeType = GetEETypeForTypeDesc(type);
+            var eeTypeDesc = _compilation.TypeSystemContext.SystemModule.GetKnownType("System", "EETypePtr");
+            if (opCode == ILOpcode.unbox)
+            {
+                var value = _stack.Pop();
+                PushExpression(StackValueKind.ByRef, "eeType", GetEETypeForTypeDesc(type), eeTypeDesc);
+                _stack.Push(value);
+                CallRuntimeExport(_compilation.TypeSystemContext, "RhUnbox2");
```

Comment:
Does this actually get compiled in and work?

Reply:
It seems to run, looking at the implementation there doesn't appear to be any reason for it not to. Somewhat of a side question, how do you run wasm for results? when I load the produced html file in firefox I can watch for trap! but I never get any of the console output.

Reply:
I generate html via ```tests\runtest /mode wasm``` and then open tests\src\Simple\HelloWasm\bin\Debug\x64\native\HelloWasm.html in the latest version of Firefox. If you have the Fall Creators Update of Windows 10, you can also open it in Edge. If those don't work, you could try running ```emrun  tests\src\Simple\HelloWasm\bin\Debug\x64\native\HelloWasm.html``` which will launch a local web server. You should be able to open the address it prints in Firefox, Chrome, or Edge (Fall Creators Update version).

Reply:
odd, that's how I run it. I just cant see any output (other than trap!)

Reply:
I switched the test to throw on a bad value, I think the unbox is right, but the box isn't allocating enough memory for the header + data.

Reply:
I remembered that older versions of emscripten had problems like that. Did you make sure to follow the update instructions at https://kripken.github.io/emscripten-site/docs/getting_started/downloads.html#updating-the-sdk . I think you need 1.37 or higher but 1.35 is the latest full installer so you have to update from there.